exporter/elasticexporter/internal/translator/elastic/traces.go

```diff
@@ -0,0 +1,467 @@
+			&transaction, &transactionContext,
+		); err != nil {
+			return err
+		}
+		transaction.Context = transactionContext.modelContext()
+		w.RawString(`{"transaction":`)
+		if err := transaction.MarshalFastJSON(w); err != nil {
+			return err
+		}
+		w.RawString("}\n")
+	} else {
+		span := model.Span{
+			ID:        spanID,
+			TraceID:   traceID,
+			ParentID:  parentID,
+			Timestamp: model.Time(startTime),
+			Duration:  durationMillis,
+			Name:      name,
+		}
+		if err := setSpanProperties(otlpSpan, &span); err != nil {
+			return err
+		}
+		w.RawString(`{"span":`)
+		if err := span.MarshalFastJSON(w); err != nil {
+			return err
+		}
+		w.RawString("}\n")
+	}
+
+	// TODO(axw) we don't currently support sending arbitrary events
+	// to Elastic APM Server. If/when we do, we should also transmit
+	// otlpSpan.TimeEvents. When there's a convention specified for
+	// error events, we could send those.
+
+	return nil
+}
+
+func setTransactionProperties(
+	otlpSpan *otlptrace.Span,
+	otlpLibrary *otlpcommon.InstrumentationLibrary,
+	tx *model.Transaction, context *transactionContext,
+) error {
+	var (
+		netHostName string
+		netHostPort int
+		netPeerIP   string
+		netPeerPort int
+	)
+
+	for _, attr := range otlpSpan.GetAttributes() {
+		switch k := attr.GetKey(); k {
+		// http.*
+		case conventions.AttributeHTTPMethod:
+			context.setHTTPMethod(attr.GetStringValue())
+		case conventions.AttributeHTTPURL:
+			if err := context.setHTTPURL(attr.GetStringValue()); err != nil {
+				return err
+			}
+		case conventions.AttributeHTTPTarget:
+			if err := context.setHTTPURL(attr.GetStringValue()); err != nil {
+				return err
+			}
+		case conventions.AttributeHTTPHost:
+			if err := context.setHTTPHost(attr.GetStringValue()); err != nil {
+				return err
+			}
+		case conventions.AttributeHTTPScheme:
+			context.setHTTPScheme(attr.GetStringValue())
+		case conventions.AttributeHTTPStatusCode:
+			context.setHTTPStatusCode(int(attr.GetIntValue()))
+		case conventions.AttributeHTTPFlavor:
+			context.setHTTPVersion(attr.GetStringValue())
+		case conventions.AttributeHTTPServerName:
+			context.setHTTPHostname(attr.GetStringValue())
+		case conventions.AttributeHTTPClientIP:
+			context.setHTTPRequestHeader("X-Forwarded-For", attr.GetStringValue())
+		case conventions.AttributeHTTPUserAgent:
+			context.setHTTPRequestHeader("User-Agent", attr.GetStringValue())
+		case "http.remote_addr":
+			// NOTE(axw) this is non-standard, sent by opentelemetry-go's othttp.
+			// It's semantically equivalent to net.peer.ip+port.
+			stringValue := attr.GetStringValue()
+			ip, port, err := net.SplitHostPort(stringValue)
+			if err != nil {
+				ip = stringValue
+			}
+			if net.ParseIP(ip) != nil {
+				netPeerIP = ip
+				netPeerPort, _ = strconv.Atoi(port)
+			}
+
+		// net.*
+		case conventions.AttributeNetPeerIP:
+			netPeerIP = attr.GetStringValue()
+		case conventions.AttributeNetPeerPort:
+			netPeerPort = int(attr.GetIntValue())
```

Comment:
In case `http.remote_addr` and `AttributeNetPeerXXX` are set it is not defined which value takes precedence over the other. (Not a big deal probably, but might be worth to check if the value is already set).

Reply:
I've added a check, but I expect this code will be deleted sooner or later.